### PR TITLE
fix: align `QueryInterface` type export with value exports

### PR DESCRIPTION
**Pull request by evanrittenhouse**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change
Fix the export/typing issues pointed out in [this comment](https://github.com/sequelize/sequelize/pull/15339#discussion_r1045288006).

Definitely a hacky solution, so if there's no better one I'd like to revisit this later. I'm not super familiar with our import/export system and how typings interact across the codebase, so any feedback is welcome

### CODE REVIEW

**Review by ephys — COMMENTED** [review]
I don't know what the best solution would be for this, I can't make up my mind :(

Options:

1. Name it `QueryInterface` everywhere:
   - 🟩 Good for our TSDoc, since it's going to reference the same name they can import
   - 🟧 Inconsistent with how we name our other QueryInterface classes & other Abstract classes
2. Name it `AbstractQueryInterface` everywhere
   - 🟩 Good for our TSDoc, since it's going to reference the same name they can import
   - 🟩 Consistent with how we name our other QueryInterface classes & other "Base dialect" classes
   - 🟧 User might be put off by the "Abstract" prefix? IDK
3. Use a different name inside of Sequelize and outside of Sequelize
   - 🟧 Bad for our TSDoc, since the Type in our TSDoc will not reflect what the user can import

@sequelize/maintainers if you have any input

**Comment by WikiRik** [comment]
Option 2 is what I would go for. `Abstract` is a term we use throughout the codebase so it should not be difficult to explain to users

File: src/sequelize.js [hunk 1]
```diff
@@ -1280,7 +1280,8 @@ Sequelize.prototype.Validator = Sequelize.Validator = Validator;
 
 Sequelize.Model = Model;
 
-Sequelize.QueryInterface = AbstractQueryInterface;
+const QueryInterface = AbstractQueryInterface;
+Sequelize.QueryInterface = QueryInterface;
```
Comment (WikiRik) [inline src/sequelize.js:1284]: So in my opinion the only change that's needed (out of the several that are now done in this PR) is;
```suggestion
Sequelize.AbstractQueryInterface = AbstractQueryInterface;
```

**Comment by ephys** [comment]
We'll need to document that breaking change in our upgrade guide, but I'm ok with it

**Comment by evanrittenhouse** [comment]
@WikiRik Where else do we use it in a user facing context?

**Comment by WikiRik** [comment]
> @WikiRik Where else do we use it in a user facing context? I'm just worrying if that's more of an implementation detail, and we may want to have the actual property on `Sequelize` be `QueryInterface`. But if that's involved elsewhere in user-facing contexts specifically, I'm good with it as well.

I thought we had more, but it seems that we only have this type export;
https://github.com/sequelize/sequelize/blob/a47af85de694c40142bd84f61bf5ec583ee9679d/src/index.d.ts#L40

File: src/index.mjs [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const and = Pkg.and;
 export const or = Pkg.or;
 
 // export * from './lib/query-interface';
-export const QueryInterface = Pkg.QueryInterface;
+export const AbstractQueryInterface = Pkg.QueryInterface;
```
Comment (WikiRik) [inline src/index.mjs:21]:
```suggestion
export const AbstractQueryInterface = Pkg.AbstractQueryInterface;
```

Comment (evanrittenhouse) [reply src/index.mjs]: I see - I need to give more of a look at how dual-mode exports work. Seems like at a high level the `index.d.ts` file has to match `index.mjs` though?

Comment (ephys) [reply src/index.mjs]: `Sequelize.AbstractQueryInterface` is the export of the class, and it's added as a static property on the Sequelize class for legacy reasons, which is that the Sequelize class is the default export:

```ts
// this works
const Sequelize = require('@sequelize/core');

// therefore, doing this gets the "AbstractQueryInterface" static property from the Sequelize class
const { AbstractQueryInterface } = require('@sequelize/core');

// This also works and is the preferred way of importing Sequelize.
const { Sequelize } = require('@sequelize/core');
```

`queryInterface: AbstractQueryInterface` is an instance property on the Sequelize class. It's not an export but the current instance of QueryInterface

**Review by WikiRik — DISMISSED** [review]
Could you open a PR at the website repo to document this changed export?

File: src/index.mjs (hunk 2 above)
Comment (evanrittenhouse) [reply src/index.mjs]: Makes sense - thank you!

**Review by ephys — APPROVED** [review]
Could you document this breaking change [in the upgrade guide](https://sequelize.org/docs/v7/other-topics/upgrade/#deprecations--removals)?

**Pull request merged**